### DISCUSSION TITLE
Unterschiedliche Polling Zeiten.

### DISCUSSION THREAD

**Discussion by schmimat** [philippoo66/optolink-splitter#16] [Ideas]
Hallo,

erst mal Danke für den Aufwand und die seht gute Umsetzung des Optolink-splitters.
Ich verwende momentan vcontrold und ioBroker. Das möchte ich allerdings durch deinen Optolink-splitter ersetzen, um beim Empfangen der Nachrichten (MQTT) im ioBroker flexibler zu sein, als mit dem Viessmann Adapter.

Dazu hätte ich noch eine Frage, Anregung. Ist es Möglich, unterschiedliche Poling-Intervalle für verschiedene Abfragepunkte zu setzen? Hast du hier eine Idee? (Datum 1 x am Tage, Leistung alle 2 Sekunden)

Vielen Dank nochmal für das tolle Projekt und das Teilen deiner Arbeit.

Mit freundlichen Grüßen Matthias

**Comment by philippoo66** [comment]
Moin Matthias!

Danke für das Lob und das Interesse!

Man könnte natürlich in der Polling Liste bsw. noch einen Eintrag für ein 'zusätzliches Intervall' rein machen und im Programm ein Int Array, wo der Zähler hochgezäht wird. Aber Z.B. bei dem Datum möchte man ja, dass es mit der ersten Abfrage am neuen Tag gelesen wird. Auch das könnte man realisieren, bei der Datei-Schreib-Pufferung sind ja schon derartige Mechanismen eingebaut.

ABER eigentlich 'kostet' es ja nix, wenn man die Sachen öfter ausliest. Das Vitoconnect 'orgelt' ja auch ständig alles durch. Vielleicht könnte es Sinn machen, MQTT nur zu schicken, wenn sich was geändert hat?! Das ginge ja auch recht einfach mit Puffer und Vergleich am besten im MQTT Modul.

Wo du dir schon Gedanken um Rationalität machst - Es steht ja noch als TODO im Code, zu überlegen, ob man das Vitoconnect 'in die Reihe einordnet', also nicht previligiert behandelt (zur Zeit immer 1 Vitoconnect Request, 1 der 3 anderen). Benutzt du das Vitoconnect? Hättest du Lust, auszuprobieren, ob das auch noch problemlos funktioniert, wenn man es nicht previligiert behandelt?

Und noch ne Frage: Mit vcontrold werden ja 'Klartext Commands' benutzt. Ich habe im Hinterkopf, die Polling Liste und vlt eine ergänzende 'Klartext Liste' zu benutzen, damit man statt `read;0x800;2;0.1;true` einfach ein "read;AussenTemp" schicken könnte (und dabei die Einzelheiten aus der Polling-/ergänzenden List genommen wird). Meinst du das lohnt den Aufwand, oder wird ein 'manuelles Lesen/Schreiben' eh eher weniger benutzt?

Grüsse!
Phil

**Reply by Huskerdu24** [reply]
Ich würde beide Weiterentwicklungen begrüßen. Besonders die Klartext-Liste. Sie würde die Arbeit mit den Partametern mMn vereinfachen, z.B beim Anlegen von Sensoren in Home Assistant.
